tests/components/demo/test_update.py

```diff
@@ -0,0 +1,153 @@
+    assert state.attributes[ATTR_TITLE] == "Awesomesoft Inc."
+    assert state.attributes[ATTR_CURRENT_VERSION] == "1.0.0"
+    assert state.attributes[ATTR_LATEST_VERSION] == "1.0.1"
+    assert (
+        state.attributes[ATTR_RELEASE_SUMMARY] == "Awesome update, fixing everything!"
+    )
+    assert state.attributes[ATTR_RELEASE_URL] == "https://www.example.com/release/1.0.1"
+
+    state = hass.states.get("update.demo_no_update")
+    assert state
+    assert state.state == STATE_OFF
+    assert state.attributes[ATTR_TITLE] == "AdGuard Home"
+    assert state.attributes[ATTR_CURRENT_VERSION] == "1.0.0"
+    assert state.attributes[ATTR_LATEST_VERSION] == "1.0.0"
+    assert state.attributes[ATTR_RELEASE_SUMMARY] is None
+    assert state.attributes[ATTR_RELEASE_URL] is None
+
+    state = hass.states.get("update.demo_add_on")
+    assert state
+    assert state.state == STATE_ON
+    assert state.attributes[ATTR_TITLE] == "AdGuard Home"
+    assert state.attributes[ATTR_CURRENT_VERSION] == "1.0.0"
+    assert state.attributes[ATTR_LATEST_VERSION] == "1.0.1"
+    assert (
+        state.attributes[ATTR_RELEASE_SUMMARY] == "Awesome update, fixing everything!"
+    )
+    assert state.attributes[ATTR_RELEASE_URL] == "https://www.example.com/release/1.0.1"
+
+    state = hass.states.get("update.demo_living_room_bulb_update")
+    assert state
+    assert state.state == STATE_ON
+    assert state.attributes[ATTR_TITLE] == "Philips Lamps Firmware"
+    assert state.attributes[ATTR_CURRENT_VERSION] == "1.93.3"
+    assert state.attributes[ATTR_LATEST_VERSION] == "1.94.2"
+    assert state.attributes[ATTR_RELEASE_SUMMARY] == "Added support for effects"
+    assert (
+        state.attributes[ATTR_RELEASE_URL] == "https://www.example.com/release/1.93.3"
+    )
+    assert state.attributes[ATTR_DEVICE_CLASS] == UpdateDeviceClass.FIRMWARE
+
+    state = hass.states.get("update.demo_update_with_progress")
+    assert state
+    assert state.state == STATE_ON
+    assert state.attributes[ATTR_TITLE] == "Philips Lamps Firmware"
+    assert state.attributes[ATTR_CURRENT_VERSION] == "1.93.3"
+    assert state.attributes[ATTR_LATEST_VERSION] == "1.94.2"
+    assert state.attributes[ATTR_RELEASE_SUMMARY] == "Added support for effects"
+    assert (
+        state.attributes[ATTR_RELEASE_URL] == "https://www.example.com/release/1.93.3"
+    )
+    assert state.attributes[ATTR_DEVICE_CLASS] == UpdateDeviceClass.FIRMWARE
+
+
+async def test_update_with_progress(hass: HomeAssistant) -> None:
+    """Test update with progress."""
+    state = hass.states.get("update.demo_update_with_progress")
+    assert state
+    assert state.state == STATE_ON
+    assert state.attributes[ATTR_IN_PROGRESS] is False
+
+    events = []
+    hass.helpers.event.async_track_state_change_event(
+        "update.demo_update_with_progress", callback(lambda event: events.append(event))
+    )
+
+    with patch("homeassistant.components.demo.update.asyncio.sleep") as fake_sleep:
```

Comment:
Patch the time slept instead of patching asyncio. Break out the time to sleep into a constant to make the patching easy.